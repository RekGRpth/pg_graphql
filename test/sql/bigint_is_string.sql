begin;

    create table blog_post(
        id bigserial primary key,
        title text not null,
        parent_id bigint references blog_post(id)
    );


    select graphql.resolve($$
    mutation {
      createBlogPost(object: {
        title: "hello"
        parentId: "1"
      }) {
        id
        parentId
      }
    }
    $$);

    select graphql.resolve($$
    mutation {
      updateBlogPostCollection(set: {
        title: "xx"
      }) {
        affectedCount
        records {
          id
          parentId
        }
      }
    }
    $$);

    select graphql.resolve($$
    {
      blogPostCollection {
        totalCount
        edges {
          node {
            id
            parentId
            parent {
              id
              parentId
            }
          }
        }
      }
    }
    $$);

rollback;
