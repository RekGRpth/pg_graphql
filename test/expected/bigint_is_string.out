begin;
    create table blog_post(
        id bigserial primary key,
        title text not null,
        parent_id bigint references blog_post(id)
    );
    select graphql.resolve($$
    mutation {
      createBlogPost(object: {
        title: "hello"
        parentId: "1"
      }) {
        id
        parentId
      }
    }
    $$);
                                 resolve                                  
--------------------------------------------------------------------------
 {"data": {"createBlogPost": {"id": "1", "parentId": "1"}}, "errors": []}
(1 row)

    select graphql.resolve($$
    mutation {
      updateBlogPostCollection(set: {
        title: "xx"
      }) {
        affectedCount
        records {
          id
          parentId
        }
      }
    }
    $$);
                                                        resolve                                                        
-----------------------------------------------------------------------------------------------------------------------
 {"data": {"updateBlogPostCollection": {"records": [{"id": "1", "parentId": "1"}], "affectedCount": 1}}, "errors": []}
(1 row)

    select graphql.resolve($$
    {
      blogPostCollection {
        totalCount
        edges {
          node {
            id
            parentId
            parent {
              id
              parentId
            }
          }
        }
      }
    }
    $$);
                                                                           resolve                                                                            
--------------------------------------------------------------------------------------------------------------------------------------------------------------
 {"data": {"blogPostCollection": {"edges": [{"node": {"id": "1", "parent": {"id": "1", "parentId": "1"}, "parentId": "1"}}], "totalCount": 1}}, "errors": []}
(1 row)

rollback;
